graphql-webclient/src/main/java/graphql/kickstart/spring/webclient/boot/GraphQLWebClientImpl.java

```diff
@@ -35,8 +35,13 @@
   }
 
   @Override
-  public <T> Mono<T> post(GraphQLRequest<T> request) {
-    return execute(request).map(it -> readValue(it, request.getReturnType()));
+  public Mono<GraphQLResponse> post(GraphQLRequest<?> request) {
+    WebClient.RequestBodySpec spec = webClient.post()
```

Comment:
Instead of copy pasting this section from the `execute` method, extract this into a separate method and reuse it in both places.